crates/trident/src/subsystems/osconfig/mod.rs

```diff
@@ -39,9 +41,20 @@ fn os_config_requires_os_modifier(ctx: &EngineContext) -> bool {
         || !os_config.services.enable.is_empty()
         || !os_config.services.disable.is_empty()
         || !os_config.kernel_command_line.extra_command_line.is_empty()
+        || !os_config.sysexts.is_empty()
+        || !os_config.confexts.is_empty()
         || should_carry_over_hostname(ctx)
 }
 
+/// Returns whether the given OS configuration requires the os-modifier binary
+/// to be present on a runtime update.
+fn runtime_update_os_config_requires_os_modifier(ctx: &EngineContext) -> bool {
```

Comment:
maybe add unit tests for runtime_udpate_os_config_requires_os_modifier and to cover sysext/confext linds in os_config_requires_os_modifier?